vscripts/confogl.nut

```diff
@@ -2,65 +2,308 @@
+			// If any plugin overrides this function, we return its return value.
+            return res;
+          }
+	    }
+	  }
+    }
+	return retVal;
   }
   // string GetDefaultItem(index)
   // Repeatedly called with incrementing indices. 
   // Return a string of a weapon name to make it a default item for survivors, or 0 to end the iteration.
   function GetDefaultItem(index) {
     Msg(format("DirectorOptions.GetDefaultItem(%d)\n", index));
+	// Default to ending iteration.
+    local retVal = 0;
+	for ( local idx = pluginScopes.len() - 1; idx >= 0; --idx ) {
+      local nextPlugin = pluginScopes[idx];
+	  if ("DirectorOptions" in pluginScopes[idx]) {
+	    if ("GetDefaultItem" in pluginScopes[idx].DirectorOptions) {
+          local fn = pluginScopes[idx].DirectorOptions.GetDefaultItem;
+          if ( typeof (fn) == "function" ) {
+            local res = fn(index)
+		    // If any plugin overrides this function, we return its return value.
+		    return res;
+          }
+	    }
+	  }
+    }
+	// We don't want to take away the default single pistol if no plugins override this function.
+	if (index == 0) {
+	  return "weapon_pistol"
+	}
+	return retVal;
   }
   // bool ShouldAvoidItem(string classname)
   // To do: Probably a bot related function or spawn related
+  // Yep, survivor bots avoid picking up items when this returns true.
   function ShouldAvoidItem(classname) {
     Msg(format("DirectorOptions.ShouldAvoidItem(%s)\n", classname));
-    return false;
+    // Default to no avoidance.
+    local retVal = false;
+	for ( local idx = pluginScopes.len() - 1; idx >= 0; --idx ) {
+      local nextPlugin = pluginScopes[idx];
+	  if ("DirectorOptions" in pluginScopes[idx]) {
+	    if ("ShouldAvoidItem" in pluginScopes[idx].DirectorOptions) {
+          local fn = pluginScopes[idx].DirectorOptions.ShouldAvoidItem;
+          if ( typeof (fn) == "function" ) {
+            local res = fn(classname)
+		    // If any plugin returns true, we return true.
+            if(res == true) {
+              return true;
+            }
+          }
+		}
+	  }
+    }
+	return retVal;
   }
-  // bool ShouldPlayBossMusic(index) 
+  // bool ShouldPlayBossMusic(index)
+  // Returns true or false if the music index is allowed to play. 
   function ShouldPlayBossMusic(index) {
     Msg(format("DirectorOptions.ShouldPlayBossMusic(%d)\n", index));
-    return true;
+    // Default to playing boss music.
+    local retVal = true;
+	for ( local idx = pluginScopes.len() - 1; idx >= 0; --idx ) {
+      local nextPlugin = pluginScopes[idx];
+	  if ("DirectorOptions" in pluginScopes[idx]) {
+	    if ("ConvertZombieClass" in pluginScopes[idx].DirectorOptions) {
+          local fn = pluginScopes[idx].DirectorOptions.ShouldPlayBossMusic;
+          if ( typeof (fn) == "function" ) {
+            local res = fn(index)
+		    // If any plugin returns false, we return false.
+            if(res == false) {
+              return false;
+            }
+          }
+		}
```

Comment:
[nit] indentation. Let's use 2 spaces for indentation to match the existing file.